src/Mod/Part/Gui/Command.cpp

```diff
@@ -1309,6 +1309,57 @@ bool CmdPartBoolean::isActive(void)
     return (hasActiveDocument() && !Gui::Control().activeDialog());
 }
 
+/**
+ * Used for Loft, Sweep, Revolve, and Extrude where the user has preselected
+ * some objects prior to executing the command. In some such cases the dialog
+ * can be skipped, for example, selecting a sketch in the tree, a path
+ * in the 3d view, and then clicking the Sweep toolbar icon
+ */
+
+namespace PartGui {
+
+bool callFunction (const Py::Object& module, const std::string& function, App::Document* doc, const char* text) {
+    try {
+        doc -> openTransaction(text);
+        Py::Boolean result(module.callMemberFunction(function));
+        bool ok = result.as_bool();
+        ok ? doc->commitTransaction() : doc->abortTransaction();
+        return ok;
+    }
+    catch (const Py::Exception&) {
+        doc->abortTransaction();
+        throw; //re-throw the exception
+    }
+}
+
+void checkSkipDialog (Gui::TaskView::TaskDialog *dlg, const std::string& function, const char* text) {
+
+    bool neverSkipDialog = App::GetApplication().GetUserParameter().GetGroup("BaseApp/Preferences/Mod/Part")->GetBool("NeverSkipDialog", false);
+    if (!neverSkipDialog) {
+        App::Document* doc = App::GetApplication().getActiveDocument();
+        Base::PyGILStateLocker lock;
+        try {
+            PyObject* module = PyImport_ImportModule("BasicShapes.SkipGui");
+            if (!module) {
+                throw Py::Exception();

```

Comment:
Probably good to add a reason string for exception throwing.

Reply:
I think this only gets thrown if the SkipGui.py file can not be found or possibly if the lock is being held by another thread.  The reason gets shown in the catch block does it not?